tests/src/test_cases/test_mem.c

```diff
@@ -0,0 +1,40 @@
+#if LV_BUILD_TEST
+#include "../lvgl.h"
+
+#include "unity/unity.h"
+
+void setUp(void)
+{
+    /* Function run before every test */
+}
+
+void tearDown(void)
+{
+    /* Function run after every test */
+}
+
+/* Non-zero size with null pointer should behave like malloc */
+void test_realloc_safe_behave_malloc_like(void)
+{
+    void * ptr = NULL;
+    void * data = &ptr;
+    size_t allocation_size = sizeof(lv_font_t);
+
+    lv_res_t retval = lv_mem_realloc_safe(&data, allocation_size);
+    TEST_ASSERT_EQUAL_UINT16(retval, LV_RES_OK);
+
+    /* Don't leak memory */
+    lv_mem_free(ptr);
```

Comment:
You free either data or ptr, the variable you have, but not both.

Reply:
So it is ok to only free `ptr`?